wger/core/api/views.py

```diff
@@ -125,3 +137,56 @@ class WeightUnitViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = WeightUnitSerializer
     ordering_fields = "__all__"
     filter_fields = ("name",)
+
+
+class ApiCreateUser(viewsets.ModelViewSet):
+    permission_classes = (AllowAny,)
+    serializer_class = (ApiCreateUserSerializer)
+
+    def create(self, request):
+        token = request.data.get('token')
+        user_id_check = get_object_or_404(Token, key=token)
+        permission = UserProfile.objects.filter(
+            user=user_id_check.user_id, is_allowed=True)
+        if not permission:
+            return Response(
+                'You arent authorised to create user contact admin for rights',
+                status=status.HTTP_400_BAD_REQUEST)
+        serializer = self.serializer_class(data=request.data)
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        permission = UserProfile.objects.filter(
+            user=self.request.user.id, is_allowed=True)
+        if User.objects.filter(
```

Comment:
Great work @NicholusMuwonge. you have good variable namings. I have a concern here, you can do a little bit of refactoring in this class and put some conditions in a separate file which can be like a helpers file. Then import them and use them in this class. With this, we shall be able to improve our code concerning code climate issues.

Reply:
okay, let me improve on that